lib/collection/src/collection_manager/segments_searcher.rs

```diff
@@ -478,7 +478,7 @@ fn execute_batch_search(
         .map(|p| p.indexed_only)
         .unwrap_or(false);
     if ignore_plain_index
-        && check_is_indexed_condition(
+        && !is_search_optimized(
```

Comment:
this is the main change, all the rest are tests

Reply:
You beat me to it!